### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by cschadewitz**

### CODE REVIEW

**Comment by cschadewitz** [comment]
Misread portion of the checklist and didnt include the Business Tests.
Will add these on the next commit

**Comment by ybaghlaf** [comment]
I will review this

File: SecretSanta/SecretSanta.Business.Tests/UserRepositoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+using SecretSanta.Data;
+
+namespace SecretSanta.Business.Tests
+{
+    [TestClass]
+    public class UserRepositoryTests
+    {
+        Mock<User> mockValidUser = new Mock<User>();
+        Mock<User> mockEditedValidUser = new Mock<User>();
+        Mock<User> mockEditedInvalidUser = new Mock<User>();
```
Comment (Keboo) [inline SecretSanta/SecretSanta.Business.Tests/UserRepositoryTests.cs:16]: For fields we specify access modifiers.
```suggestion
        private Mock<User> mockValidUser = new Mock<User>();
        private Mock<User> mockEditedValidUser = new Mock<User>();
        private Mock<User> mockEditedInvalidUser = new Mock<User>();
```

File: SecretSanta/src/SecretSanta.Data/DataCollection.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SecretSanta.Data;
+
+namespace SecretSanta.Business
+{
+    public class DataCollection<T> : List<T> where T : IData
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Data/DataCollection.cs:10]: Very clever implementation. Threw me off at first how your indexer was returning null when the index was out of bounds.

**Review by Keboo — COMMENTED** [review]
Really great work!
- Create a UserController in the API projects
  - Implement endpoints for:
    - Retrieving all users ✔
    - Retriving a single user by id ✔
    - Creating a user ✔
    - Updating a user ✔
    - Deleting a user ✔
  - The UserController should function. ✔
  - Include a `.rest` file for interacting with the controller. ✔
- The `IUserRepository` interface is provided for you. You will need to provide an implementation of this interface within the SecretSanta.Business project. This implementation can simply store the user within a list, dictionary, array, or similar. ✔
- All test projects should be within a test directory ✔
- Create a unit test project for the SecretSanta.Business project. ✔
  - Unit test all public methods ✔
  - Ensure the project is named with a ".Tests" suffix ✔
  - Ensure test class is named appropriately for the class being tested: ✔
  - Name all tests in the form `<MethodBeingTested>_<ConditionBeingTested>_<ExpectedOutcome>()`. You can add  `<NoWarn>CA1707;INTL0003</NoWarn>` to remove the naming warnings. ✔
- Create a Unit Test project for the SecretSanta.Api project. ✔
  - Test all controller public methods. It is up to you to determine the best way to provide the controller's dependencies. ✔
  - Ensure the project is named with a ".Tests" suffix ✔

Extra Credit: ✔

**Pull request closed without merging**